From 321648622558aeade89480f49bca2ffdd13bf82b Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Wed, 26 Oct 2011 17:52:30 +0800
Subject: [PATCH 037/641] UniCore64: Add head.s and define swapper_pg_dir

Fixed error:
mm/built-in.o:(.data+0xd68): undefined reference to `swapper_pg_dir'

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile |    3 ++-
 arch/unicore64/kernel/head.S   |    2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/head.S

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index 5cac0d1..ef3ca57 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,2 +1,3 @@
 obj-y			:= tlb.o
-extra-y			:= vmlinux.lds
+head-y			:= head.o
+extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/head.S b/arch/unicore64/kernel/head.S
new file mode 100644
index 0000000..9ad1b3b
--- /dev/null
+++ b/arch/unicore64/kernel/head.S
@@ -0,0 +1,2 @@
+.globl	swapper_pg_dir
+.equ	swapper_pg_dir, 0
-- 
1.7.9.5

